File: Ambulance.BLL/Models/PersonModels/PersonBaseModel.cs
namespace Ambulance.BLL.Models.PersonModels;

public abstract class PersonBaseModel
{
    public string Name { get; set; } = null!;
    public string Surname { get; set; } = null!;
    public string? MiddleName { get; set; }
    public DateOnly? DateOfBirth { get; set; }
    public string? Gender { get; set; }
    public string? PhoneNumber { get; set; }
    public string? Email { get; set; }
    public string? ImageUrl { get; set; }
}

File: ambulance-site/src/api/models/PersonModels/PersonBaseModel.js
export class PersonBaseModel {
    constructor({
        personId = null,
        name = "",
        surname = "",
        middleName = null,
        dateOfBirth = null,
        gender = null,
        phoneNumber = null,
        email = null
    } = {}) {
        this.personId = personId;
        this.name = name;
        this.surname = surname;
        this.middleName = middleName;
        this.dateOfBirth = dateOfBirth;
        this.gender = gender;
        this.phoneNumber = phoneNumber;
        this.email = email;
    }
}